auth/client/src/test/java/org/wildfly/security/auth/client/EncryptedExpressionContextTest.java

```diff
@@ -0,0 +1,117 @@
+    }
+
+    @Test
+    public void removeCredentialStore() {
+        EncryptedExpressionContext ctx = EncryptedExpressionContext.empty().with(TEST_CRED_STORE_1_NAME, credentialStore1, config1);
+        ctx = EncryptedExpressionContext.empty().withOut(TEST_CRED_STORE_1_NAME, ctx.encryptedExpressionConfig);
+        Assert.assertNull(ctx.encryptedExpressionConfig.credentialStoreMap.get(TEST_CRED_STORE_1_NAME));
+    }
+
+    @Test
+    public void setCredentialStoreMap() {
+        Map<String, CredentialStore> credentialStoreMap = new HashMap<>();
+        credentialStoreMap.put(TEST_CRED_STORE_1_NAME, credentialStore1);
+        credentialStoreMap.put(TEST_CRED_STORE_2_NAME, credentialStore2);
+        EncryptedExpressionContext ctx = EncryptedExpressionContext.empty().with(credentialStoreMap, config1);
+        Assert.assertNotNull(ctx.encryptedExpressionConfig.credentialStoreMap);
+        Assert.assertNotNull(ctx.encryptedExpressionConfig.credentialStoreMap.get(TEST_CRED_STORE_1_NAME));
+        Assert.assertNotNull(ctx.encryptedExpressionConfig.credentialStoreMap.get(TEST_CRED_STORE_2_NAME));
+    }
+
+    @Test
+    public void setResolverAndTestEncryptionAndDecryption() {
+        EncryptedExpressionContext ctx = EncryptedExpressionContext.empty().with(TEST_CRED_STORE_1_NAME, credentialStore1, config1);
+        EncryptedExpressionConfig config = ctx.encryptedExpressionConfig;
+        EncryptedExpressionResolver resolver = new EncryptedExpressionResolver();
+        Map<String, EncryptedExpressionResolver.ResolverConfiguration> resolverConfigurationMap = new HashMap<>();
+        resolverConfigurationMap.put(TEST_RESOLVER_1_NAME, new EncryptedExpressionResolver.ResolverConfiguration(TEST_RESOLVER_1_NAME, TEST_CRED_STORE_1_NAME, "secretkey1"));
+        resolverConfigurationMap.put(TEST_RESOLVER_2_NAME, new EncryptedExpressionResolver.ResolverConfiguration(TEST_RESOLVER_2_NAME, TEST_CRED_STORE_2_NAME, "secretkey1"));
+        resolver.setResolverConfigurations(resolverConfigurationMap);
+        resolver.setPrefix("ENC")
+                .setResolverConfigurations(resolverConfigurationMap)
+                .setDefaultResolver(TEST_RESOLVER_1_NAME);
+        ctx = ctx.with(resolver, config);
+        Assert.assertEquals(resolver, ctx.encryptedExpressionConfig.encryptedExpressionResolver);
+        Assert.assertEquals(resolverConfigurationMap, ctx.encryptedExpressionConfig.encryptedExpressionResolver.getResolverConfiguration());
+
+        String clearText = "password";
+        String encryptedExpression = ctx.encryptedExpressionConfig.encryptedExpressionResolver.createExpression(TEST_RESOLVER_1_NAME, clearText, ctx.encryptedExpressionConfig);
+        String decryptedExpression = ctx.encryptedExpressionConfig.encryptedExpressionResolver.resolveExpression(encryptedExpression, ctx.encryptedExpressionConfig);
+        Assert.assertEquals(clearText, decryptedExpression);
+    }
+}
+
```

Comment:
@PrarthonaPaul the tests could later include an wildfly-config.xml file that contains additional schemas that can be included there. You can find them here: https://docs.wildfly.org/30/WildFly_Elytron_Security.html#Configuring_other_clients_using_wildfly-config , there are also links to schema locations

Reply:
I have added a new test that uses encrypted expression with authentication client. 
Please let me know if there are any other clients I should add tests for. So far there are tests for parsing only, but no integrated functionality tests. Please let me know if I should add those and where I can add those. 
Thanks!

Reply:
@PrarthonaPaul  we can add an integration tests that verify that the credentials used in SASL authentication are resolved as expected and authentication pass. See for example the MaskedPasswordSaslAuthenticationTest where there is a SASL client and SASL server communication and we test that their authentication is successful. We could add similar integraiton tests that uses encrypted password on the client side

Reply:
I have added a test for this. Currently looking to see if we need a programmatic one. 
But the one for xml config should be good to go. 
Thanks @Skyllarr